### DISCUSSION TITLE
Set default resize image quality

### DISCUSSION THREAD

**Discussion by sophisma** [SixLabors/ImageSharp.Web#138] [Q&A] [answered]
First of all let me start by saying that I absolutely love this thing, you guys did a great job.
I'm using this in a web application to serve responsive content and one thing I notice is that resized content lacks a bit of sharpness.
Here is an example of it.
![blog_9](https://user-images.githubusercontent.com/2579705/107824865-1030bf80-6d7a-11eb-8ab1-4e26ab8ba0ce.jpg)
![thumbnail](https://user-images.githubusercontent.com/2579705/107824900-19219100-6d7a-11eb-81a0-d52c9cfda6c2.jpg)
You can see that on the slightly smaller image there's an overall lack of shapness.
You can check it on the green square near the window, in the small image it's a lot more blurry. the power chord on the monitor it's a lot softer as well.
I know I can pass the quality parameter in the URL, or even add it on the OnParseCommandsAsync method, but is there a way to set it on the configuration?

Edit:
I've tried to set the quality in the OnParseCommandsAsync method by doing
`c.Commands["quality"] = "100";`
but this doesn't work,
In the images above the diference between both doesn't seem that huge, I'm posting two new ones to show the issue.

![blog_6](https://user-images.githubusercontent.com/2579705/107827223-44a67a80-6d7e-11eb-8beb-73a4622db9e3.jpg)
![thumbnail (1)](https://user-images.githubusercontent.com/2579705/107827253-50923c80-6d7e-11eb-945a-b684f2527d90.jpg)

It's particularly noticeable on the Playstation logo, the line in the middle of the console and the controller buttons.

**Comment by JimBobSquarePants** [comment, marked as answer]
Hi @sophisma 

Thanks! It's really great to hear you like it!

Yep you can do this at startup when registering the services.

First create the following method. That will build you a custom configuration instance using a default quality of 91.

```c#
internal static Configuration CreateCustomConfiguration()
{
    var configuration = new Configuration(
        new PngConfigurationModule(),
        new GifConfigurationModule(),
        new BmpConfigurationModule(),
        new TgaConfigurationModule());

    // 91 is the minimum quality required to force the encoder to not use
    // 4:2:0 subsampling.
    var jpegEncoder = new JpegEncoder { Quality = 91 };
    configuration.ImageFormatsManager.SetEncoder(JpegFormat.Instance, jpegEncoder);
    configuration.ImageFormatsManager.SetDecoder(JpegFormat.Instance, new JpegDecoder());
    configuration.ImageFormatsManager.AddImageFormatDetector(new JpegImageFormatDetector());

    return configuration;
}
```

Then when registering the middleware services you can call it like this. 

```c#
services.AddImageSharp(options => options.Configuration = CreateCustomConfiguration());
```

You'll need the following namepspace declarations for the code to compile.

``` c#
using SixLabors.ImageSharp.Formats.Bmp;
using SixLabors.ImageSharp.Formats.Gif;
using SixLabors.ImageSharp.Formats.Jpeg;
using SixLabors.ImageSharp.Formats.Png;
using SixLabors.ImageSharp.Formats.Tga;
```

**Reply by sophisma** [reply]
Thank you for answering so quickly.

**Reply by JimBobSquarePants** [reply]
No worries. You might also want to experiment with keeping the quality lower but explicitly setting the `JpegEncoder.Subsample` property to `JpegSubsample.Ratio444` to see if you can preserve a smaller output size.
